### DISCUSSION TITLE
String primary key with default auto_incrementing breaks app.

### DISCUSSION THREAD

**Discussion by xodix** [SeaQL/sea-orm#609] [Q&A] [answered]
## Description

When using String as primary key "Execution Error: Failed to find inserted item" appears.

## Steps to Reproduce

Entity model:

```rust
use sea_orm::prelude::*;
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, PartialEq, DeriveEntityModel, Deserialize, Serialize)]
#[sea_orm(table_name = "clients")]
pub struct Model {
    #[sea_orm(primary_key)]
    pub id: String,
    pub name: String,
    pub surname: String,
    pub nip_prefix: i32,
    pub nip: String,
    pub state: String,
    pub postcode: String,
    pub locality: String,
    pub street: String,
    pub house_nb: i32,
}

#[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
pub enum Relation {}

impl ActiveModelBehavior for ActiveModel {}
```

actix web function:

```rust
	pub async fn add(pool: Data<DatabaseConnection>, json: Json<client::Model>) -> HttpResponse {
		let pool = pool.as_ref();

		let json: client::ActiveModel = json.0.into();

		let res = json.insert(pool).await;

		if let Err(err) = res {
			println!("{err}");
			HttpResponse::InternalServerError().body(err.to_string())
		} else {
			HttpResponse::Ok().finish()
		}
	}
```

### Expected Behavior

If 'json' model is correct it should be inserted to db and 'res' should not return Err.

### Actual Behavior

Wild "Execution Error: Failed to find inserted item" appears when 'json' model is correct.

More interestingly item is correctly inserted into database.

Problem is with retrieving the inserted item.

### Reproduces How Often

always

## Versions

├── sea-orm v0.6.0
│ ├── sea-orm-macros v0.6.0 (proc-macro)  
│ ├── sea-query v0.21.0
│ │ ├── sea-query-derive v0.2.0 (proc-macro)
│ ├── sea-strum v0.23.0
│ │ └── sea-strum_macros v0.23.0 (proc-macro)

Mysql + Windows

## Additional Information

Problem disappears when disabling auto_increment:

```rust
pub struct Model {
	#[sea_orm(primary_key, auto_increment = false)]
	pub id: String,
	pub name: String,
	pub surname: String,
	pub nip_prefix: i32,
	pub nip: String,
	pub state: String,
	pub postcode: String,
	pub locality: String,
	pub street: String,
	pub house_nb: i32,
}
```

Sorry for my poor english.

**Comment by billy1624** [comment]
Hey @xodix, welcome! I wonder why the `id` column is of String type?

**Comment by billy1624** [comment]
And, is the `id` column an auto generated column?

**Comment by xodix** [comment]
Some primary keys are meant to be representation of ID codes found in an ID card or a passport.

**Comment by xodix** [comment]
> And, is the `id` column an auto generated column?

I am working with an existing database not generated by SeaORM.

**Comment by billy1624** [comment, marked as answer]
Ok, the `id` column does not auto increment on its own. So, it normal to specify `#[sea_orm(auto_increment = false)]` on the Model.
